packages/sharing-smart-contract/contracts/DataProtectorSharing.sol

```diff
@@ -166,7 +178,15 @@ contract DataProtectorSharing is
             _workerpoolOrder.category
         );
 
-        dealid = POCO_DELEGATE.matchOrders(_appOrder, _datasetOrder, _workerpoolOrder, requestOrder);
+        if (_useVoucher) {
+            IVoucher _voucher = IVoucher(VOUCHER_HUB.getVoucher(_spender));
+            dealid = _voucher.matchOrders(_appOrder, _datasetOrder, _workerpoolOrder, requestOrder);
+        } else {
+            if (_workerpoolOrder.workerpoolprice > 0) {
+                POCO_DELEGATE.transferFrom(_spender, address(this), _workerpoolOrder.workerpoolprice);
+            }
```

Comment:
we must test this branch
- when _spender has not approved the contract => should revert the consumeProtetedData tx
- when _spender has approved the contract but account balance is insufficient => should revert the consumeProtetedData tx
- when _spender has approved the contract and account balance is insufficient => should create the deal

Reply:
Commit: 8dc6e20ed4b402225f3635cab9f3859e6be49cee